src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -536,67 +530,57 @@ FixedwingPositionControl::tecs_status_publish()
-		pos_ctrl_status.target_bearing = _npfg.targetBearing();
-		pos_ctrl_status.xtrack_error = _npfg.getTrackError();
-		pos_ctrl_status.acceptance_radius = _npfg.switchDistance(500.0f);
-
-		npfg_status.lat_accel = _npfg.getLateralAccel();
-		npfg_status.lat_accel_ff = _npfg.getLateralAccelFF();
-		npfg_status.heading_ref = _npfg.getHeadingRef();
-		npfg_status.bearing = bearing;
-		npfg_status.bearing_feas = _npfg.getBearingFeas();
-		npfg_status.bearing_feas_on_track = _npfg.getOnTrackBearingFeas();
-		npfg_status.signed_track_error = _npfg.getTrackError();
-		npfg_status.track_error_bound = _npfg.getTrackErrorBound();
-		npfg_status.airspeed_ref = _npfg.getAirspeedRef();
-		npfg_status.min_ground_speed_ref = _npfg.getMinGroundSpeedRef();
-		npfg_status.adapted_period = _npfg.getAdaptedPeriod();
-		npfg_status.p_gain = _npfg.getPGain();
-		npfg_status.time_const = _npfg.getTimeConst();
-		npfg_status.timestamp = hrt_absolute_time();
-
-		_npfg_status_pub.publish(npfg_status);
-
-	} else {
-		pos_ctrl_status.acceptance_radius = _l1_control.switch_distance(500.0f);
-	}
-
+	// TODO: ?
+	// if (_l1_control.has_guidance_updated()) {
+	// 	pos_ctrl_status.nav_bearing = _l1_control.nav_bearing();
+
+	// 	pos_ctrl_status.target_bearing = _l1_control.target_bearing();
+	// 	pos_ctrl_status.xtrack_error = _l1_control.crosstrack_error();
+	// }
+
+	pos_ctrl_status.nav_bearing = NAN;
+	pos_ctrl_status.target_bearing = NAN;
+	pos_ctrl_status.xtrack_error = NAN;
+	pos_ctrl_status.nav_bearing = bearing;
+	pos_ctrl_status.target_bearing = _npfg.targetBearing();
+	pos_ctrl_status.xtrack_error = _npfg.getTrackError();
+	pos_ctrl_status.acceptance_radius = _npfg.switchDistance(500.0f);
 	pos_ctrl_status.wp_dist = get_distance_to_next_waypoint(_current_latitude, _current_longitude,
 				  _pos_sp_triplet.current.lat, _pos_sp_triplet.current.lon);
 
 	pos_ctrl_status.yaw_acceptance = NAN;
-
-	pos_ctrl_status.timestamp = hrt_absolute_time();
-
 	pos_ctrl_status.type = _position_sp_type;
+	pos_ctrl_status.timestamp = hrt_absolute_time();
 
 	_pos_ctrl_status_pub.publish(pos_ctrl_status);
+
+
+	// publish npfg_status
+	npfg_status_s npfg_status{};
+	npfg_status.wind_est_valid = _wind_valid;
+	npfg_status.lat_accel = _npfg.getLateralAccel();
+	npfg_status.lat_accel_ff = _npfg.getLateralAccelFF();
+	npfg_status.heading_ref = _npfg.getHeadingRef();
+	npfg_status.bearing = bearing;
+	npfg_status.bearing_feas = _npfg.getBearingFeas();
+	npfg_status.bearing_feas_on_track = _npfg.getOnTrackBearingFeas();
+	npfg_status.signed_track_error = _npfg.getTrackError();
+	npfg_status.track_error_bound = _npfg.getTrackErrorBound();
+	npfg_status.airspeed_ref = _npfg.getAirspeedRef();
+	npfg_status.min_ground_speed_ref = _npfg.getMinGroundSpeedRef();
+	npfg_status.adapted_period = _npfg.getAdaptedPeriod();
+	npfg_status.p_gain = _npfg.getPGain();
+	npfg_status.time_const = _npfg.getTimeConst();
+	npfg_status.timestamp = hrt_absolute_time();
+
+	_npfg_status_pub.publish(npfg_status);
```

Comment:
Do we actually need a separate `npfg_status`? or can we merge it in to pos control status?

Reply:
maybe one or the other - if there were actually more position controllers that have common outputs.. the generic name makes sense. but for the moment.. maybe just directly `npfg_status` stays.. and we kill `pos_control_status`? it has informative fields for this specific controller.

Reply:
Hey, @tstastny ans @Jaeyoung-Lim. If you don't mind, i would leave this as is for the moment. When we split up the lateral guidance and add the course, acceleration, etc. interface, we can consolidate them there and split them up as appropriate for the sub components.